pymeasure/instruments/hp/hp437b.py

```diff
@@ -0,0 +1,949 @@
+                errors.append(err)
+            else:
+                break
+        return errors
+
+    event_status = Instrument.measurement(
+        "*ESR?",
+        """
+        Get the status byte and Master Summary Status bit.
+
+        .. code-block:: python
+
+            print(instr.request_service_conditions)
+            StatusRegister.PowerOn|CommandError
+        """,
+        get_process=lambda v: EventStatusRegister(int(v))
+    )
+
+    def activate_auto_range(self):
+        """
+        The power meter divides each sensor’s power range into 5 ranges of
+        10 dB each. Range 1 is the most sensitive (lowest power levels), and
+        Range 5 is the least sensitive (highest power levels). Range 5 can be
+        less than 10 dB if the sensor’s power range is less than 50 dB. The
+        range can be set either automatically or manually.
+        'activate_auto_range' automatically selects the correct range for the current
+        measurement.
+        """
+        self.write("RA")
+
+    def calibrate(self, calibration_factor):
+        """
+        Calibrate a sensor to the power meter with a 'calibration_factor' in percent.
+        """
+        self.write("CL%.1fPCT" % calibration_factor)
+
+    @property
+    def calibration_factor(self):
+        """
+        Control the calibration factor of a specific power sensor at a specific input frequency.
+        (A chart or table of CAL FACTOR % versus Frequency is printed on each sensor and an
+        accompanying data sheet.) Calibration factor is entered in percent.
+        Valid entries for 'calibration_factor' range from 1.0 to 150.0%.
+        """
+        self.write("KB")
+        # returns CALFAC 097.9%
+        display_content = self.display_output
+        assert display_content[0:6] == "CALFAC"
+        self.write("EX")
+        return float(display_content[7:12])
+
+    @calibration_factor.setter
+    def calibration_factor(self, calibration_factor):
+        values = [1.0, 150.0]
+        strict_range(float(calibration_factor), values)
+
+        self.write("KB%3.1fPCT" % float(calibration_factor))
+        self.check_errors()
+
+    display_enabled = Instrument.setting(
+        "%s",
+        """
+        Set the display of the power meter active or inactive.
+        """,
+        map_values=True,
+        values={True: "DE", False: "DD"}
+    )
+
+    display_all_segments_enabled = Instrument.setting(
+        "%s",
+        """
+        Set all segments of the display of the power meter active or resume normal state.
+        """,
+        map_values=True,
+        values={True: "DA", False: "DE"}
+    )
+
+    display_user_message = Instrument.setting(
+        "DU %s",
+        """
+        Set a custom user message up to 12 alpha-numerical chars. If the string is empty or None
+        the user message gets disabled.
+        """,
+        validator=lambda x, y: x if str(x).isalnum() and len(str(x)) <= 12 else str(x)[0:11],
```

Comment:
`str(x)[0:11]` will return the first 11 chars, not the twelfth (at index 11) one. So you should use `[0:12]` or `[:12]` (a not written start index defaults to 0) if you want to use the first 12 chars.

Reply:
Thanks